### ISSUE TITLE
Merge TimeSeriesQueries.jl into TimeSeries

### ISSUE THREAD

**Issue by milktrader** [JuliaStats/TimeSeries.jl#288]
The (TimeSeriesQueries)[https://github.com/milktrader/TimeSeriesQueries.jl] unregistered package provides SQL-like query syntax of TimeSeries TimeArray objects. Passing tests and documentation is complete. 

Here is a quick (screencast)[https://asciinema.org/a/9zxrxfj8gca52gxppysz2vlhi]

**Comment by femtotrader** [comment]
When https://github.com/JuliaStats/TimeSeries.jl/issues/292 will be closed It should be possible to use https://github.com/davidanthoff/Query.jl

**Comment by milktrader** [comment]
I don't think so but if you find Query.jl supports TimeArrays please advise with an example.

**Comment by ararslan** [comment]
cc @davidanthoff

**Comment by davidanthoff** [comment]
You can now query ``TimeArray``s with [Query.jl](https://github.com/davidanthoff/Query.jl), and also materialize query results into a ``TimeArray``. Here are some examples:
````julia
using Query, TimeSeries, DataFrames

dates  = collect(Date(1999,1,1):Date(2000,12,31))
mytime = TimeArray(dates, rand(length(dates)))

df = @from i in mytime begin
    @group i.value by year(i.timestamp) into g
    @select {Year=g.key, Average=mean(g)}
    @collect DataFrame
end
````
If your final ``@select`` statement creates something that can be turned into a ``TimeArray`` (i.e. it has a ``timestamp`` column, and all the other columns need to have the same type), you can also materialize things into a ``TimeArray``:
````julia
ta = @from i in mytime begin                              
    @group i.value by year(i.timestamp) into g           
    @select {timestamp=Date(g.key,1,1), average=mean(g)} 
    @collect TimeArray                                   
end                                  
````
I'll also add another way to materialize into a ``TimeArray`` soon, see davidanthoff/IterableTables.jl#27. With that you will no longer have to create a ``timestamp`` column in the query.

All of this comes as a side-effect of [IterableTables.jl](https://github.com/davidanthoff/IterableTables.jl), which enables a whole bunch of other scenarios for ``TimeArray`` as well: you can convert it into lots of other table types, you can convert from lots of other table types (as long as the source table has a ``timestamp`` column and all the other columns are of one type) and you can use ``TimeArray``s with things like ``Gadfly``, ``VegaLite``, ``GLM`` etc.

Any feedback is welcome!

**Comment by davidanthoff** [comment]
Oh, one last thing: the implementation for the ``TimeArray`` sink is terrible right now, so that is known to be slow and I'll fix it at some point (davidanthoff/IterableTables.jl#11). That should only affect anything that creates a ``TimeArray``, querying one should not be affected by this.
